payjoin/src/receive/v2.rs

```diff
@@ -323,10 +323,10 @@ impl Enrolled {
 ///
 /// If you are implementing an interactive payment processor, you should get extract the original
 /// transaction with extract_tx_to_schedule_broadcast() and schedule, followed by checking
-/// that the transaction can be broadcast with check_can_broadcast. Otherwise it is safe to
+/// that the transaction can be broadcast with check_broadcast_suitability. Otherwise it is safe to
 /// call assume_interactive_receive to proceed with validation.
 pub struct UncheckedProposal {
-    inner: super::UncheckedProposal,
+    pub inner: super::UncheckedProposal,
```

Comment:
Why make this public? Ideally these typestate machines are kept opaque so that the only modifications happen through the machine.